mobile/status.go

```diff
@@ -80,38 +97,69 @@ func initializeApplication(requestJSON string) string {
 		return makeJSONResponse(err)
 	}
 
+	// Initialize logs
```

Comment:
Untested suggestion, but this log initialization section could be a separate function. Usually catches my attention whenever I see comments describing the *what* because the more useful comments are about the *why* and *what* comments can often times be made irrelevant by good function names and good function composition. Unfortunately in Go, function composition can be cumbersome without some abstractions.

```go
func initLogs(request requests.InitializeApplication) error {
	if request.LogDir == "" {
		request.LogDir = request.DataDir
	}
	logSettings := logutils.LogSettings{
		Enabled: request.LogEnabled,
		Level:   request.LogLevel,
		File:    path.Join(request.LogDir, api.DefaultLogFile),
	}

	var err error
	if err = logutils.OverrideRootLoggerWithConfig(logSettings); err == nil {
		logutils.ZapLogger().Info("logging initialised",
			zap.Any("logSettings", logSettings),
			zap.Bool("APILoggingEnabled", request.APILoggingEnabled),
		)
	} else {
		return err
	}

	if request.APILoggingEnabled {
		logRequestsFile := path.Join(request.LogDir, api.DefaultAPILogFile)
		return requestlog.ConfigureAndEnableRequestLogging(logRequestsFile)
	}

	return nil
}
```

```go
err = initLogs(request)
if err != nil {
	return makeJSONResponse(err)
}
```